test/elastic_record/model/joining_test.rb

```diff
@@ -0,0 +1,88 @@
+require 'helper'
+
+class ElasticRecord::Model::JoiningTest < MiniTest::Test
+  class Child < ActiveRecord::Base
+    include ElasticRecord::Model
+    self.elastic_index.mapping[:properties] = ::Widget.elastic_index.mapping[:properties].dup
+    self.table_name = 'widgets'
+    belongs_to :parent, foreign_key: :warehouse_id
+  end
+
+  class Parent < ActiveRecord::Base
```

Comment:
Maybe just name these classes `Mom` and `Son`, to make the tests simpler?

Reply:
Fixed.  It caught a bug--I had to add `demodulize` when inferring the relation name from the klass.